app/src/main/java/com/pileproject/drive/execution/ExecutionCondition.java

```diff
@@ -17,48 +17,186 @@
 package com.pileproject.drive.execution;
 
 
-import android.content.Context;
-
-import com.pileproject.drive.R;
-import com.pileproject.drive.app.DriveApplication;
 import com.pileproject.drive.programming.visual.block.BlockBase;
+import com.pileproject.drive.programming.visual.block.repetition.RepetitionEndBlock;
+import com.pileproject.drive.programming.visual.block.repetition.WhileForeverBlock;
+import com.pileproject.drive.programming.visual.block.selection.SelectionEndBlock;
 
 import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Stack;
 
 /**
- * This class has the condition of execution
- *
- * @author <a href="mailto:tatsuyaw0c@gmail.com">Tatsuya Iwanari</a>
- * @version 1.0 7-July-2013
+ * A container class that has the condition of program execution.
  */
 public class ExecutionCondition {
-    public static final int TRUE = 1;
-    public static final int FALSE = 0;
-    public Stack<Integer> whileStack;
-    public Stack<Map<String, Integer>> ifStack;
-    public int beginningOfCurrentWhileLoop;
-    public int programCount;
-    public ArrayList<BlockBase> blocks;
-
-    public ExecutionCondition() {
-        whileStack = new Stack<>();
-        ifStack = new Stack<>();
-        beginningOfCurrentWhileLoop = -1;
+    private Stack<Integer> mWhileStack;
+    private Stack<IfStatus> mIfStack;
+    private int mBeginningOfCurrentLoop;
+    private int mProgramCount;
+    private ArrayList<BlockBase> mBlocks;
+
+    public ExecutionCondition(ArrayList<BlockBase> blocks) {
```

Comment:
Finally, I have decided to change it to `List` according to your comments. But let me leave a note for this. The interface of List does not specify the time complexity of `get(int index)` and this program needs random accesses frequently. That's why I have used ArrayList not other List. But the complexity can be ignored because the size of the list is small enough in practical.